packages/legends/src/types.ts

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { SymbolProps } from './svg/symbols/types'
+import { CompleteTheme } from '@nivo/core'
+
+/**
+ * This can be used to add effect on legends on interaction.
+ */
+type EffectProps = {
+    on: 'hover'
+    style: Partial<{
+        itemTextColor: string
+        itemBackground: string
+        itemOpacity: number
+        symbolSize: number
+        symbolBorderWidth: number
+        symbolBorderColor: string
+    }>
+}
+
+type SymbolShape = 'circle' | 'diamond' | 'square' | 'triangle'
+
+type BoxLegendSymbolProps = Partial<{
+    symbolShape: SymbolShape | React.FC<SymbolProps>
+    symbolSize: number
+    symbolSpacing: number
+    symbolBorderWidth: number
+    symbolBorderColor: string
+}>
+
+type InteractivityProps = Partial<
+    Record<
+        'onClick' | 'onMouseEnter' | 'onMouseLeave',
+        (datum: Datum, event: React.MouseEvent<SVGRectElement>) => void
+    >
+>
+
+export type LegendAnchor =
+    | 'top'
+    | 'top-right'
+    | 'right'
+    | 'bottom-right'
+    | 'bottom'
+    | 'bottom-left'
+    | 'left'
+    | 'top-left'
+    | 'center'
+
+export type LegendDirection = 'column' | 'row'
+
+export type LegendItemDirection =
+    | 'left-to-right'
+    | 'right-to-left'
+    | 'top-to-bottom'
+    | 'bottom-to-top'
+
+export type Datum = {
+    id: string | number
+    label: string | number
+    color?: string
+    fill?: string
+}
+
+type CommonLegendProps = {
+    direction: LegendDirection
+    padding: number | Record<'top' | 'right' | 'bottom' | 'left', number>
+    justify?: boolean
+
+    itemWidth: number
+    itemHeight: number
+    itemDirection?: LegendItemDirection
+    itemTextColor?: string
+    itemBackground?: string
+    itemOpacity?: number
+    itemsSpacing?: number
+
+    effects?: EffectProps[]
+}
+
+export type LegendProps = {
+    data: Datum[]
```

Comment:
Is requiring a Datum[] here right? I can't find anything in the [Legends guide](https://nivo.rocks/guides/legends) or in the storybooks demonstrating the use of this or explaining what it's for. This and the required `padding` above both broke every one of my  charts.